components/style/properties/helpers/animated_properties.mako.rs

```diff
@@ -2002,34 +2005,22 @@ impl Animate for Perspective {
     }
 }
 
-impl Animate for MatrixDecomposed3D {
-    /// <https://drafts.csswg.org/css-transforms/#interpolation-of-decomposed-3d-matrix-values>
-    fn animate(&self, other: &Self, procedure: Procedure) -> Result<Self, ()> {
+impl MatrixDecomposed3D {
+    fn slerp(from: &Quaternion, to: &Quaternion, procedure: Procedure) -> Result<Quaternion, ()> {
```

Comment:
Make this an implementation of `Animatable` for `Quaternion` instead of keeping it on `MatrixDecomposed3D` .